src/main/scala/io/iohk/ethereum/domain/branch/BestBlockchainBranch.scala

```diff
@@ -0,0 +1,37 @@
+package io.iohk.ethereum.domain.branch
+import akka.util.ByteString
+
+import io.iohk.ethereum.db.storage.BlockNumberMappingStorage
+import io.iohk.ethereum.domain.Block
+import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.domain.BlockchainReader
+
+/** A Branch instance which only works for the best canonical branch. As this branch
+  * currently has specific indexes (particularly regarding accessing blocks by number),
+  * it will uses thoses to provide better performance.
+  */
+class BestBlockchainBranch(
+    tipBlockHeader: BlockHeader,
+    bestChainBlockNumberMappingStorage: BlockNumberMappingStorage,
+    blockchainReader: BlockchainReader
+) extends BlockchainBranch {
+
+  /* The following assumptions are made in this class :
+   *  - the whole branch exist in storage
+   *  - The various metadata and index are consistent
+   *
+   * If those assumptions are found to be false, then the application is in an inconsistent
+   * state and the class will throw.
```

Comment:
throw which error exaclty?

Reply:
In fact this comment might not be relevant anymore. It wrote it while trying to put too many function at once inside this class. You're right I should define a dedicated Exception if I econter the case though. I will remove the comment for now.